std-bits/table/src/main/java/org/enso/table/format/xlsx/Reader.java

```diff
@@ -187,32 +198,201 @@ private static Table read_table(
+    for (int i = builders.size(); i <= columnCount; i++) {
+      Builder builder = new InferredBuilder(size);
+      builder.appendNulls(rows);
+      builders.add(builder);
+    }
+  }
+
+  private static int getSheetIndex(Workbook workbook, String sheetName) {
+    int sheetCount = workbook.getNumberOfSheets();
+    for (int i = 0; i < sheetCount; i++) {
+      if (workbook.getSheetName(i).equalsIgnoreCase(sheetName)) {
+        return i;
+      }
+    }
+    return -1;
   }
 
-  private static Object getCellValue(
-      Cell cell, CellType cellType, Function<LocalDate, Value> mkDate) {
+  private static Object getCellValue(Cell cell) {
+    CellType cellType = cell.getCellType();
+    if (cellType == CellType.FORMULA) {
+      cellType = cell.getCachedFormulaResultType();
+    }
+
     switch (cellType) {
-      case FORMULA:
-        return getCellValue(cell, cell.getCachedFormulaResultType(), mkDate);
       case NUMERIC:
         if (DateUtil.isCellDateFormatted(cell)) {
-          return mkDate.apply(cell.getLocalDateTimeCellValue().toLocalDate());
+          return cell.getLocalDateTimeCellValue().toLocalDate();
         } else {
           return cell.getNumericCellValue();
         }
       case STRING:
         return cell.getStringCellValue();
-      case ERROR:
-        return null;
       case BOOLEAN:
         return cell.getBooleanCellValue();
+      case ERROR:
       case BLANK:
-        return null;
       case _NONE:
         return null;
     }
     return null;
   }
+
+  public static String[] readSheetNames(InputStream stream, boolean xls_format) throws IOException {
+    Workbook workbook = xls_format ? new HSSFWorkbook(stream) : new XSSFWorkbook(stream);
+    int sheetCount = workbook.getNumberOfSheets();
+    var output = new String[sheetCount];
+    for (int i = 0; i < sheetCount; i++) {
+      output[i] = workbook.getSheetName(i);
+    }
+    return output;
+  }
+
+  public static String[] readRangeNames(InputStream stream, boolean xls_format) throws IOException {
+    Workbook workbook = xls_format ? new HSSFWorkbook(stream) : new XSSFWorkbook(stream);
+    return workbook.getAllNames().stream().map(Name::getNameName).toArray(String[]::new);
+  }
+
+  public static Table readSheetByName(
+      InputStream stream,
+      String sheetName,
+      Integer skip_rows,
+      Integer row_limit,
+      boolean xls_format)
+      throws IOException, IllegalArgumentException {
+    Workbook workbook = xls_format ? new HSSFWorkbook(stream) : new XSSFWorkbook(stream);
+
+    int sheetIndex = getSheetIndex(workbook, sheetName);
+    if (sheetIndex == -1) {
+      throw new IllegalArgumentException("Unknown sheet '" + sheetName + "'.");
+    }
+
+    return readSheetToTable(
+        workbook,
+        sheetIndex,
+        null,
+        skip_rows == null ? 0 : skip_rows,
+        row_limit == null ? Integer.MAX_VALUE : row_limit);
+  }
+
+  public static Table readSheetByIndex(
```

Comment:
Not sure but I think it may be worth adding some short docs to the public API. I know it is internal to us, but still it's a border between Java and Enso and may be helpful to have this documented.